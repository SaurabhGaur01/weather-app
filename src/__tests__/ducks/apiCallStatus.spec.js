import {
    setApiCallStatusError,
    setApiCallStatusSuccess,
    setApiCallStatusLoading,
    apiCallStatusReducer,
} from '../../ducks/apiCallStatus';

import { 
    API_CALL_STATUS_SUCCESS, API_CALL_STATUS_ERROR, API_CALL_STATUS_LOADING
} from '../../constants/status';

describe('apiCallStatus Duck Tests', () => {
    const mockState = {
        key: 'mockKey',
    }

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setApiCallStatusLoading should match the snapshot' , () => {
            expect(setApiCallStatusLoading()).toMatchSnapshot();
        });
        it('setApiCallStatusError should match the snapshot' , () => {
            expect(setApiCallStatusError()).toMatchSnapshot();
        });
        it('setApiCallStatusSuccess should match the snapshot' , () => {
            expect(setApiCallStatusSuccess()).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(apiCallStatusReducer()).toEqual('');
        });
        it('should return the passed state if called with no action' , () => {
            expect(apiCallStatusReducer(mockState)).toBe(mockState);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(apiCallStatusReducer(mockState, unknownAction)).toBe(mockState);
        });
    });

    describe('when called with a particular action', () => {
        it('should return the value of the constant API_CALL_STATUS_SUCCESS' , () => {
            expect(apiCallStatusReducer(mockState, setApiCallStatusSuccess()))
            .toEqual(API_CALL_STATUS_SUCCESS);
        });
        it('should return the value of the constant API_CALL_STATUS_ERROR' , () => {
            expect(apiCallStatusReducer(mockState, setApiCallStatusError()))
            .toEqual(API_CALL_STATUS_ERROR);
        });
        it('should return the value of the constant API_CALL_STATUS_LOADING' , () => {
            expect(apiCallStatusReducer(mockState, setApiCallStatusLoading()))
            .toEqual(API_CALL_STATUS_LOADING);
        });
    });

})