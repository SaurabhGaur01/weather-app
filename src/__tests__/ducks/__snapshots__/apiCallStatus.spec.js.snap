// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`apiCallStatus Duck Tests Action Generator setApiCallStatusError should match the snapshot 1`] = `
Object {
  "type": "modules/weather-app/SET_API_CALL_STATUS_ERROR",
}
`;

exports[`apiCallStatus Duck Tests Action Generator setApiCallStatusLoading should match the snapshot 1`] = `
Object {
  "type": "modules/weather-app/SET_API_CALL_STATUS_LOADING",
}
`;

exports[`apiCallStatus Duck Tests Action Generator setApiCallStatusSuccess should match the snapshot 1`] = `
Object {
  "type": "modules/weather-app/SET_API_CALL_STATUS_SUCCESS",
}
`;
